src/witty/compile_module.py

```diff
@@ -47,6 +52,224 @@ def get_witty_cache_dir() -> Path:
+    extra_link_args: list[str] | None = None,
+    force_rebuild: bool | None = None,
+    quiet: bool = False,
+    output_dir: Path | None = None,
+    **extension_kwargs: Any,
+) -> ModuleType:
+    """Compile a nanobind C/C++ module given as a string.
+
+    The module will be stored in witty's cache directory. Called with the same
+    `source`, the cached module will be returned.
+
+    Parameters
+    ----------
+    source: str
+        The C/C++ source code.
+    source_files : list of Path, optional
+        Additional source files the C/C++ code depends on. Changes to these
+        files will trigger re-compilation of the module.
+    include_dirs : list of Path, optional
+        List of directories to search for C/C++ header files (in Unix
+        form for portability).
+    library_dirs : list of Path, optional
+        List of directories to search for C/C++ libraries at link time.
+    language : str, optional
+        Extension language (i.e., "c", "c++", "objc"). Defaults to "c++".
+    extra_compile_args : list of str, optional
+        Extra platform- and compiler-specific information to use when
+        compiling the source files in 'sources'. This is typically a
+        list of command-line arguments for platforms and compilers where
+        "command line" makes sense.
+    extra_link_args : list of str, optional
+        Extra platform- and compiler-specific information to use when
+        linking object files to create the extension (or a new static
+        Python interpreter). Has a similar interpretation as for 'extra_compile_args'.
+    force_rebuild : bool, optional
+        Force a rebuild even if a module with the same name/hash already exists. By
+        default False.  May be set via environment variable `WITTY_FORCE_REBUILD=1`.
+    quiet : bool, optional
+        Suppress output except for errors and warnings.
+    output_dir : Path, optional
+        Directory to store the compiled module. If not provided, the module will be
+        stored in the output of `witty.get_witty_cache_dir()`:
+        - os.environ["WITTY_CACHE_DIR"] if set
+        - Windows: `%LOCALAPPDATA%/witty/cache`
+        - macOS: `~/Library/Caches/witty`
+        - Linux: os.environ['XDG_CACHE_HOME']/witty or `~/.cache/witty`
+    extension_kwargs : dict, optional
+        Additional keyword arguments passed to the distutils `Extension` constructor.
+
+    Returns
+    -------
+    ModuleType
+        The compiled module.
+    """
+
+    def build_extra_objects(build_dir: Path):
+        nanobind_base_dir = Path(nanobind.include_dir()) / ".."
+        source_dir = Path(nanobind.source_dir())
+        target = build_dir / str(source_dir).lstrip("/") / "nb_combined.o"
+
+        if target.exists():
+            return [str(target)]
+
+        compiler = new_compiler()
+
+        compiler.add_include_dir(nanobind.include_dir())
+        compiler.add_include_dir(nanobind_base_dir / "ext" / "robin_map" / "include")
```

Comment:
```suggestion
        compiler.add_include_dir(
            str(nanobind_base_dir / "ext" / "robin_map" / "include")
        )
```

(not all versions of python gracefully convert Path to str)